packages/block-editor/src/hooks/block-rename-ui.js

```diff
@@ -0,0 +1,170 @@
+/**
+ * WordPress dependencies
+ */
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { addFilter } from '@wordpress/hooks';
+import { __ } from '@wordpress/i18n';
+import { getBlockSupport } from '@wordpress/blocks';
+import {
+	MenuItem,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+	Button,
+	TextControl,
+	Modal,
+} from '@wordpress/components';
+import { useState } from '@wordpress/element';
+import { useDispatch } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../store';
+import {
+	BlockSettingsMenuControls,
+	useBlockDisplayInformation,
+} from '../components';
+
+const notEmptyString = ( testString ) => testString?.trim()?.length > 0;
+
+function RenameModal( { blockName, onClose, onSave } ) {
+	const [ editedBlockName, setEditedBlockName ] = useState( blockName );
+
+	const nameHasChanged = editedBlockName !== blockName;
+
+	const isNameValid = nameHasChanged && notEmptyString( editedBlockName );
+
+	return (
+		<Modal title={ __( 'Rename block' ) } onRequestClose={ onClose }>
+			<p>{ __( 'Choose a custom name for this block.' ) }</p>
+			<form className="sidebar-navigation__rename-modal-form">
+				<VStack spacing="3">
+					<TextControl
+						__nextHasNoMarginBottom
+						value={ editedBlockName }
+						placeholder={ __( 'Block name' ) }
+						onChange={ setEditedBlockName }
+					/>
+					<HStack justify="right">
+						<Button variant="tertiary" onClick={ onClose }>
+							{ __( 'Cancel' ) }
+						</Button>
+
+						<Button
+							disabled={ ! isNameValid }
```

Comment:
This should be `aria-disabled`.

Reply:
Question - I assume I'm then doing to need to implement all the styling affordances and submission prevention right?

Reply:
The editor has the same styling for `aria-disabled` and `disabled`. The latter isn't great for a11y as the button won't receive to focus.

This only leaves the submission prevention.

Reply:
✅